tests/integration/targets/ec2/tasks/main.yml

```diff
@@ -0,0 +1,184 @@
+
+    - name: create a subnet to use for creating an ec2 instance
+      ec2_vpc_subnet:
+        az: '{{ availability_zone }}'
+        tags: '{{ resource_prefix }}_setup'
+        vpc_id: '{{ setup_vpc.vpc.id }}'
+        cidr: '{{ subnet_cidr }}'
+        state: present
+        resource_tags:
+          Name: '{{ resource_prefix }}_setup'
+      register: setup_subnet
+
+    - name: create a security group to use for creating an ec2 instance
+      ec2_group:
+        name: '{{ resource_prefix }}_setup'
+        description: 'created by Ansible integration tests'
+        state: present
+        vpc_id: '{{ setup_vpc.vpc.id }}'
+      register: setup_sg
+
+    - name: Find AMI to use
+      ec2_ami_info:
+        owners: 'amazon'
+        filters:
+          name: '{{ ec2_ami_name }}'
+      register: ec2_amis
+
+    - name: Set fact with latest AMI
+      vars:
+        latest_ami: '{{ ec2_amis.images | sort(attribute="creation_date") | last }}'
+      set_fact:
+        ec2_ami_image: '{{ latest_ami.image_id }}'
+
+    # ============================================================
+
+    - name: test first instance is started
+      ec2:
+        instance_type: t2.micro
+        key_name: '{{ setup_key.key.name }}'
+        state: present
+        image: '{{ ec2_ami_image }}'
+        wait: yes
+        instance_tags:
+          ResourcePrefix: '{{ resource_prefix }}-integration_tests'
+        group_id: '{{ setup_sg.group_id }}'
+        vpc_subnet_id: '{{ setup_subnet.subnet.id }}'
+      register: test_instance_1
+
+    - name: test second instance is started
+      ec2:
+        instance_type: t2.micro
+        key_name: '{{ setup_key.key.name }}'
+        state: present
+        image: '{{ ec2_ami_image }}'
+        wait: yes
+        instance_tags:
+          ResourcePrefix: '{{ resource_prefix }}-another_tag'
+        group_id: '{{ setup_sg.group_id }}'
+        vpc_subnet_id: '{{ setup_subnet.subnet.id }}'
+      register: test_instance_2
+
+    - name: assert instances started
+      assert:
+        that:
+          - "test_instance_1.instances[0].state == 'running'"
+          - "test_instance_2.instances[0].state == 'running'"
+
+    - name: test first instance is terminated
+      ec2:
+        instance_ids: "{{ test_instance_1.instance_ids }}"
+        state: absent
+        wait: yes
+      register: result
+
+    - name: assert instance terminated
+      assert:
+        that:
+          - "result.instances[0].state == 'terminated'"
+
+    - name: test terminated instance is ignored when stopping
+      ec2:
+        instance_tags:
+          ResourcePrefix: '{{ resource_prefix }}-integration_tests'
+        state: stopped
+        wait: yes
+      register: result
+
+    - name: assert resource not changed
+      assert:
+        that:
+          - "result.changed == False"
+
+    - name: assert second instance still running
+      assert:
+        that:
+          - "test_instance_2.instances[0].state == 'running'"
```

Comment:
The `test_instance_2` var is not updated after the stop action on `test_instance_1`; you're asserting against the value that was set at line 92.  For this type of assertion you'll need to use `community.aws.ec2_instance_info` to query all instances based on tag of `ResourcePrefix: '{{ resource_prefix }}-*`, then assert how many there are and what what their status are.

Reply:
Is there a reason you suggest querying based on the tag instead of based on the instance id? With instance id it seems like I'd know exactly what I'm asserting against.

Reply:
In this case you could totally use id with ec2_instance_info to get the current state.  Broadly, tags should be unique per test run if they're using resource_prefix, so querying for the tag will get you "whatever the matching instances are at this point in time".  That can be useful in tests where you're creating and tearing down resources repeatedly and want to be sure you've got the correct, current view of what things look like in the correct var.  Either works in this specific case, so long as we're updating the var before asserting something about it, just something to keep in mind when manipulating instances.